dashboard/src/module/example/index.js

```diff
@@ -0,0 +1,25 @@
+// @flow
+
+import * as React from 'react';
+
+import {register} from '@sensu/lib/runtime';
+
+import Panel from '@sensu/lib/component/Panel';
+import Button from '@sensu/lib/component/Button';
+
+class ExampleRouteHandler extends React.PureComponent<mixed> {
+  render() {
+    return <Panel>Example <Button>Shared Component</Button></Panel>;
+  }
+}
+
+register({
+  id: 'some:module:id',
+  routes: [
+    {
+      path: '/example',
+      component: ExampleRouteHandler,
+    },
+  ],
+  navigation: [{label: 'Example Route', location: 'example'}],
+});
```

Comment:
If registration is a side-effect of evaluating the module could we encounter issues where a route / component isn't yet available and we display a 404 or encounter an exception?

🤔 Might be nice if static details of a module are available before the module itself has been loaded / evaluated. Could also open the door for loading modules on-demand.

Reply:
This is totally dependent on script execution order. This concern is avoided if we ensure the `client` bundle is loaded last.

If an extension is registered while a 404 view is currently rendered, the extension may seamlessly take over the route.

Reply:
I'm thinking through how the extension manifest could be defined and served as static JSON, but this presents issues with linking to defined React components without leaking into global scope.

Reply:
🤔 Perhaps each handler component could be defined and registered at runtime with a scoped ID and referenced in a purely static manifest.